### ISSUE TITLE
Cannot import name 'astral_event' from 'custom_components.sun2.helpers' 

### ISSUE THREAD

**Issue by bigmoby** [pnbruckner/ha-sun2#40]
Logger: homeassistant.components.hassio
Source: components/hassio/__init__.py:482
Integration: Home Assistant Supervisor (documentation, issues)

Platform error sensor.sun2 - cannot import name 'astral_event' from 'custom_components.sun2.helpers' (/config/custom_components/sun2/helpers.py)

Some other information about the environment:
- Sun2 version: 2.0.0
- Home Assistant OS: 2021.5.0

**Comment by pnbruckner** [comment]
I tested this configuration and it seems to work fine. Are you sure you installed/upgraded this custom integration properly?

**Comment by bigmoby** [comment]
> I tested this configuration and it seems to work fine. Are you sure you installed/upgraded this custom integration properly?

Thank you so much for your help and time 🙏
mmm yes, I'm pretty sure it's installed properly: I copy/paste from GitHub raw view to pre-existing sun2 files (older version). I copy/paste them via visual editor plugin.

**Comment by bigmoby** [comment]
Tomorrow morning (now in Italy is late) I could try by git clone and give you a feedback

**Comment by bigmoby** [comment]
Unfortunately also from git clone (so we could exclude any copy/paste mistake) the problem still remains. I attach two screen shot in order to better understand and define the issue (filelist.md5 is a checksum md5 in order to demonstrate the right file contents itself):
<img width="628" alt="Screenshot 2021-05-07 at 10 10 54" src="https://user-images.githubusercontent.com/11453002/117421741-17054380-af1f-11eb-8f7c-b26513b05a30.png">
<img width="413" alt="Screenshot 2021-05-07 at 10 13 09" src="https://user-images.githubusercontent.com/11453002/117421744-18367080-af1f-11eb-9d5d-0c6ec780d454.png">
<img width="570" alt="Screenshot 2021-05-07 at 10 26 04" src="https://user-images.githubusercontent.com/11453002/117421745-18cf0700-af1f-11eb-80a8-057b33072ebd.png">

**Comment by MihleTX** [comment]
I have the same error showing up after migrating to 5.0.  I double checked and re-copied all of the most recent 5 files into custom components and have the exact same error that @bigmoby has.

**Comment by MihleTX** [comment]
OK - I went back and re-copied all the 5 files, and copied the example full configuration.  This time I am not getting the same error, but none of the sensors or binary sensors are being produced...  I will restart HA again to see if it continues to happen.

**Comment by pnbruckner** [comment]
@bigmoby I'm not sure what to tell you. I can verify those are the correct md5 sums. It works for me on 2021.5.0. What does your sun2 related config look like?

Is it some sort of permission problem? I notice the files are all owned by root (although they're readable by anyone.) Normally, once HA runs, there would be a `__pycache__` folder (in `custom_components/sun2`), but I don't see one in your case.
```text
phil@GGHWFS1:~/repos/ha-sun2/custom_components/sun2 (master)$ ls -lah
total 60K
drwxrwxr-x 3 phil phil 4.0K May  5 15:52 .
drwxrwxr-x 3 phil phil 4.0K Jun 19  2019 ..
-rw-rw-r-- 1 phil phil  13K May  5 15:52 binary_sensor.py
-rw-rw-r-- 1 phil phil 2.3K May  5 15:52 helpers.py
-rw-rw-r-- 1 phil phil   24 Jun 19  2019 __init__.py
-rw-rw-r-- 1 phil phil  289 May  5 15:52 manifest.json
drwxrwxr-x 2 phil phil 4.0K May  6 14:33 __pycache__
-rw-rw-r-- 1 phil phil  17K May  5 15:52 sensor.py
```

@MihleTX, that's weird. (BTW, what is "5.0"?) Did you restart HA after copying the files? Also, what was your sun2 related config (when it didn't work and before copying the "full example" config)?

**Comment by bigmoby** [comment]
> 
> Is it some sort of permission problem? I notice the files are all owned by root (although they're readable by anyone.) Normally, once HA runs, there would be a `__pycache__` folder (in `custom_components/sun2`), but I don't see one in your case.

Hi @pnbruckner thanks a lot again for your support on this issue!
Yes, I deleted the `__pycache__` directory before the md5 function in order to have a clean picture of the environment...of course that directory has been created in the precompilation process (when the flow fails). 👍
Now...I'll try to fix the owner for the files but I think it's not related to the issue because at the first run (when I copy/paste from editor) I suppose the rights haven't changed and only during the second tentative (by git clone from shell) the rights and the owner changed because of shell user process. Anyway I'll try this way! 👍

Thank you @MihleTX I really appreciate your step guide! The re-copy process sounds me weird because...mmm...I don't know but appear to me a "non deterministic" case, I don't know how could explain that. But by your guide I imagine that I have to control the platform configuration if some parameter changes between different versions. 🤷‍♂️ 
Thank you very much guys, I'll inform you if I'll fix!

**Comment by bigmoby** [comment]
Ok guys, I solved...how?! I don't know! 😅
Really! Let me explain better...
For the permission I noticed that all files created by visual editor are owned by root, so I haven't changed the owner but only the permission flags (a 755 chow)...but the problem remains! 
After that I shuffled the code in helpers.py (yes I know...an idiot last chance desperate action!) and pasted astral event function at the beginning of the file...and of course nothing changed! 😅
In the meantime an update comes from Home Assistant and I updated it moving on 2021.5.1 and...after reboot and several warning messages about something of "sun2 is not supported by home assistant...etc..." all the components regarding sun2 run ok! 👍...so the problem was...boh!!! 😉
Thank you again guys!
All the best
Fabio Mauro 🙌
